orttraining/orttraining/test/python/perf_log/ort_module_perf_test_tools.py

```diff
@@ -0,0 +1,206 @@
+    'sequence_length']
+
+def calculate_and_log_perf_metrics(args, start_time,
+    mysql_server_name, power_bi_user_name, power_bi_password, power_bi_database, ort_repo_path=None):
+    completion_time = datetime.datetime.now()
+    perf_metrics_duration = completion_time - start_time
+
+    for attribute in required_attributes_for_perf_metrics:
+        if not hasattr(args, attribute):
+            raise ValueError('args does not contain all attributes needed to calculate perf metrics. \
+                Please prepare perf_metrics and call log_perf_metrics instead')
+
+    perf_metrics = {}    
+    perf_metrics['Model'] = args.model_name
+    perf_metrics['BatchId'] = 'NA'
+    perf_metrics['ModelName'] = args.model_name
+    perf_metrics['DisplayName'] = args.model_name
+    perf_metrics['UseMixedPrecision'] = args.fp16 if hasattr(args, 'fp16') else False
+    perf_metrics['UseAutoCast'] = args.use_auto_cast if hasattr(args, 'use_auto_cast') else False
+    perf_metrics['UseDeepSpeed'] = args.use_deep_speed if hasattr(args, 'use_deep_speed') else False
+    perf_metrics['Optimizer'] = args.optimizer
+    perf_metrics['BatchSize'] = args.batch_size
+    perf_metrics['SeqLen'] = args.sequence_length
+    perf_metrics['PredictionsPerSeq'] = args.prediction_per_seq if hasattr(args, 'prediction_per_seq') else 0
+    perf_metrics['NumOfBatches'] = args.epochs * args.train_steps
+    perf_metrics['WeightUpdateSteps'] = args.epochs * args.train_steps
+    perf_metrics['Round'] = 0                                       # NA
+    perf_metrics['GradAccSteps'] = args.gradient_accumulation_steps
+
+    perf_metrics['AvgTimePerBatch'] = \
+        perf_metrics_duration.microseconds / args.train_steps
+
+    perf_metrics['Throughput'] = \
+        args.batch_size * args.train_steps / perf_metrics_duration.seconds
+
+    perf_metrics['StabilizedThroughput'] = 0    # TODO
+    perf_metrics['EndToEndThroughput'] = 0      # TODO
+    perf_metrics['TotalTime'] = perf_metrics_duration.seconds
+
+    perf_metrics['AvgCPU'] = 0                  # TODO
+    perf_metrics['Memory'] = 0                  # TODO
+    perf_metrics['RunConfig'] = 'na'
+    perf_metrics['Time'] = completion_time.strftime("%Y-%m-%d %H:%M:%S")
+
+    log_perf_metrics(perf_metrics, mysql_server_name, power_bi_user_name, power_bi_password, power_bi_database,
+        ort_repo_path)
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--mysql_server_name', help='Perf dashboard mysql server name')
+    parser.add_argument('--power_bi_user_name', help='Power BI user name')
+    parser.add_argument('--password', help='password', default=None)
+    parser.add_argument('--database', help='The dashboard database')
+    return parser.parse_args()
+
+def ConnectAndInsertPerfMetrics(mysql_server_name, power_bi_user_name, password, database, perf_metrics):
+    conn = ConnectToPerfDashboardDb(mysql_server_name, power_bi_user_name, password, database)
```

Comment:
Please follow PEP 8 style for function names: "Function names should be lowercase, with words separated by underscores as necessary to improve readability" (instead of `ConnectAndInsertPerfMetrics`
and `ConnectToPerfDashboardDb`).

Reply:
name changed